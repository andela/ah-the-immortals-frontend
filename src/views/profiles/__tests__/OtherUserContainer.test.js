import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import thunk from 'redux-thunk';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import ViewUserProfile from '../OtheUserContainer';

describe('Test view user profile container', () => {
  let testStore = configureMockStore([thunk]);
  let store = testStore(
    {
      Profile: {profile: {} }
    });

  
  const wrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <ViewUserProfile 
          match={{params: {profile: {}}, isExact: true, path: '', url: ''}}
        />
      </BrowserRouter>
    </Provider>
  );

  it('test view user profile container snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('simulates handle follow', () => {
    const followBtn = wrapper.findWhere(n => n.type() === 'button');
    followBtn.simulate('click', {
      preventDefault: jest.fn()
    });
    expect(followBtn.length).toEqual(1);
  });
});
