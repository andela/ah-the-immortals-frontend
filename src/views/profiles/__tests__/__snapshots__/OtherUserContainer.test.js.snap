// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test view user profile container test view user profile container snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(ViewUserProfile)
        match={
          Object {
            "isExact": true,
            "params": Object {
              "profile": Object {},
            },
            "path": "",
            "url": "",
          }
        }
      >
        <ViewUserProfile
          followAction={[Function]}
          getProfile={[Function]}
          match={
            Object {
              "isExact": true,
              "params": Object {
                "profile": Object {},
              },
              "path": "",
              "url": "",
            }
          }
          profile={Object {}}
          unfollowAction={[Function]}
        >
          <UserProfileView
            bio=""
            firstName=""
            following="Follow"
            handleFollow={[Function]}
            image=""
            lastName=""
            username=""
          >
            <div>
              <Container
                className="profile"
                fluid={false}
              >
                <div
                  className="profile container"
                >
                  <div
                    className="container-fluid"
                  >
                    <div
                      className="container"
                    >
                      <div
                        className="card profile-card"
                      >
                        <div
                          className="centered"
                        >
                          <div
                            className="float-left profile-avatar"
                          >
                            <img
                              alt="Avatar"
                              className="prof-avatar"
                              src="test-file-stub"
                            />
                          </div>
                        </div>
                        <h4>
                          @
                          username [ ]
                        </h4>
                        <p>
                          This is my biography
                        </p>
                        <div>
                          <Link
                            to="/userprofile/"
                          >
                            <a
                              href="/userprofile/"
                              onClick={[Function]}
                            >
                              <Button
                                active={false}
                                className="btn primary"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                                variant="primary"
                              >
                                <button
                                  className="btn primary btn btn-primary"
                                  disabled={false}
                                  onClick={[Function]}
                                  type="button"
                                >
                                  Follow
                                </button>
                              </Button>
                            </a>
                          </Link>
                        </div>
                      </div>
                    </div>
                    <ul
                      className="nav nav-tabs"
                    >
                      <li
                        className="nav-item"
                      >
                        <a
                          className="nav-link active"
                          data-toggle="tab"
                          href="#home"
                        >
                          <b>
                             3 
                          </b>
                          Articles
                        </a>
                      </li>
                    </ul>
                    <div
                      className="tab-content profile-content-articles"
                      id="myTabContent"
                    >
                      <div
                        className="tab-pane fade active show"
                        id="home"
                      >
                        <div
                          className="row"
                        >
                          <div
                            className="col"
                          >
                            <div
                              className="card"
                            >
                              <img
                                alt="article"
                                className="card-img-top"
                                src="https://mdbootstrap.com/img/Photos/Others/images/43.jpg"
                              />
                              <div
                                className="card-body"
                              >
                                <h4
                                  className="card-title"
                                >
                                  <a
                                    href="#card"
                                  >
                                    Article title
                                  </a>
                                </h4>
                                <p
                                  className="card-text"
                                >
                                  Once Leonidas realizes he will be surrounded, he sends away the Greek allies to alert the cities to the south. Being too few to hold the pass, the Spartans instead attack the Persian front, where Xerxes is nearby.
                                </p>
                                <a
                                  className="btn btn-primary"
                                  href="#view"
                                >
                                  View
                                </a>
                              </div>
                            </div>
                          </div>
                          <div
                            className="col"
                          >
                            <div
                              className="card"
                            >
                              <img
                                alt="article"
                                className="card-img-top"
                                src="https://mdbootstrap.com/img/Photos/Others/images/43.jpg"
                              />
                              <div
                                className="card-body"
                              >
                                <h4
                                  className="card-title"
                                >
                                  <a
                                    href="#card"
                                  >
                                    Article title
                                  </a>
                                </h4>
                                <p
                                  className="card-text"
                                >
                                  Once Leonidas realizes he will be surrounded, he sends away the Greek allies to alert the cities to the south. Being too few to hold the pass, the Spartans instead attack the Persian front, where Xerxes is nearby.
                                </p>
                                <a
                                  className="btn btn-primary"
                                  href="#view"
                                >
                                  View
                                </a>
                              </div>
                            </div>
                          </div>
                          <div
                            className="col"
                          >
                            <div
                              className="card"
                            >
                              <img
                                alt="article"
                                className="card-img-top"
                                src="https://mdbootstrap.com/img/Photos/Others/images/43.jpg"
                              />
                              <div
                                className="card-body"
                              >
                                <h4
                                  className="card-title"
                                >
                                  <a
                                    href="#card"
                                  >
                                    Article title
                                  </a>
                                </h4>
                                <p
                                  className="card-text"
                                >
                                  Once Leonidas realizes he will be surrounded, he sends away the Greek allies to alert the cities to the south. Being too few to hold the pass, the Spartans instead attack the Persian front, where Xerxes is nearby.
                                </p>
                                <a
                                  className="btn btn-primary"
                                  href="#view"
                                >
                                  View
                                </a>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </Container>
              <Footer>
                <footer
                  className="bg-light text-dark mt-4 fixed-bottom"
                >
                  <div
                    className="container-fluid py-2"
                  >
                    <div
                      className="row"
                    >
                      <div
                        className="col"
                      >
                        <span
                          className="small"
                        >
                          <br />
                          <Link
                            to="#null"
                          >
                            <a
                              href="/#null"
                              onClick={[Function]}
                            >
                              Terms and Conditions
                            </a>
                          </Link>
                        </span>
                      </div>
                      <div
                        className="col text-center"
                      >
                        <br />
                        Authors Haven
                         
                      </div>
                      <div
                        className="col text-right small align-self-end"
                      >
                        Copyright ©
                         
                        2019
                         
                        Immortals
                      </div>
                    </div>
                  </div>
                </footer>
              </Footer>
            </div>
          </UserProfileView>
        </ViewUserProfile>
      </Connect(ViewUserProfile)>
    </Router>
  </BrowserRouter>
</Provider>
`;
